components/renku_data_services/session/k8s_client.py

```diff
@@ -105,6 +105,12 @@ async def delete_build_run(self, name: str) -> None:
             logger.exception(f"Cannot delete build {name} because of {e}")
         return None
 
+    async def cancel_build_run(self, name: str) -> BuildRun:
+        """Cancel a Shipwright BuildRun."""
+        build = await ShipwrightBuildRunV1Beta1Kr8s.get(name=name, namespace=self.namespace)
+        await build.patch({"spec": {"state": "BuildRunCanceled"}})
```

Comment:
question: do we need to check if the build exists here?

Reply:
If the buildrun does not exist, then this will return an error from the `.get()` call.

Note that this should not happen because this is called just after the status of the buildrun has been refreshed.